modules/packages/Collection.chpl

```diff
@@ -18,118 +18,212 @@
  */
 
 /*
-  A 'Collection' is a data structure, a container for elements that provide support
-  for insert, lookup, remove, and iteration operations.
+  
+  Summary
+  _______
+
+  A 'Collection' is an interface for a data structure that is a 'contract' between
+  the user and implementer that it fulfills the following:
+
+  1. Is safe parallel-safe, hence is safe to use across multiple tasks across multiple locales.

```

Comment:
I am not sure if Collection provides parallel-safety guarantee as it is now. Nor I think it should. i.e. I think Collection can be inherited by a data structure that is not parallel-safe.

Reply:
Eh... I guess relaxing this property means more Collections, sure. I can't wait until we get proper interfaces and inheritance of said interfaces though.